feat(kpop): component reskin [KHCP-8994] (#2176)
Kong/kongponents: docs/components/popover.md

# Popover

KPop is a popover component that comes in handy when you need to display more content than a tooltip can fit.

## Props

### buttonText

> Popover trigger button text. If you want to use your custom element as popover trigger, check out [`default` slot](#default).

Reviewer: ```suggestion
Popover trigger button text. If you want to use your custom element as a popover trigger, check out the [`default` slot](#default).
```

Whether popover should be opened on trigger element click or mouseover.

Accepted values are:
- `click` (default)
- `hover`

<KPop
  trigger="hover"
  button-text="Open popover"
>
  <template #content>
    Popover content.
  </template>
</KPop>

```html
<KPop
  trigger="hover"
  button-text="Open popover"
>
  <template #content>
    Popover content.
  </template>
</KPop>
```

### popoverTimeout

When [`trigger` prop](#trigger) is `hover`, you can provide a timeout for popover to wait before it closes. Default value is 300 milliseconds.

<KPop
  :popover-timeout="3000"
  trigger="hover"
  button-text="Open popover"
>
  <template #content>
    Popover content.
  </template>
</KPop>

```html
<KPop
  :popover-timeout="3000"
  trigger="hover"
  button-text="Open popover"
>
  <template #content>
    Popover content.
  </template>
</KPop>
```

### disabled

Boolean to control whether popover should be disabled. Defaults to `false`.

<KPop
  disabled
  button-text="Open popover"
>
  <template #content>
    Popover content.
  </template>
</KPop>

```html
<KPop
  disabled
  button-text="Open popover"
>
  <template #content>
    Popover content.
  </template>
</KPop>
```

### hideCaret

> Boolean to control whether the popover caret should be visible. Defaults to `true`.

Reviewer: Are you sure? I feel like the examples in the doc are defaulting to the caret being displayed, not hidden.

![image](https://github.com/Kong/kongponents/assets/67973710/0a88db1b-f558-4eb5-8f37-2e93245accad)

No `:hide-caret="false"` and caret is visible

Author: Sorry I'm confused. Are you suggesting that hideCaret prop is not working as expected? Default value is `false` - so **do show caret** unless told otherwise. IN the prop example I am hiding the caret and that works as expected too.
![Screenshot 2024-05-10 at 1 01 08 PM](https://github.com/Kong/kongponents/assets/36751160/61501c90-5f2d-4ec4-919e-50038cc52f01)

Reviewer: ```suggestion
Boolean to control whether the popover caret should be visible. Defaults to `false`.
```

Yeah, sorry 😅 I meant to call out the text

<KPop
  hide-caret
  button-text="Open popover"
>
  <template #content>
    Popover content.
  </template>
</KPop>

```html
<KPop
  hide-caret
  button-text="Open popover"
>
  <template #content>
    Popover content.
  </template>
</KPop>
```

### closeOnPopoverClick

> Boolean to control whether or not popover should close on popover content click. Default to `false`.

Reviewer: ```suggestion
Boolean to control whether or not the popover should close when a user clicks within the popover content. Default to `false`.
```

    <KButton>button</KButton>
    <template #content>
      I am placed on the {{ selectedPosition }}!
    </template>
  </KPop>
</template>

<script lang="ts">
import { defineComponent } from 'vue'

export default defineComponent({
  data () {
    return {
      selectedPosition: 'auto',
      positions: [
        'auto',
        'top',
        'topStart',
        'topEnd',
        'left',
        'leftStart',
        'leftEnd',
        'right',
        'rightStart',
        'rightEnd',
        'bottom',
        'bottomStart',
        'bottomEnd'
      ]
    }
  }
})
</script>
```

### positionFixed

Use fixed positioning of the popover to avoid content being clipped by parental boundaries - defaults to `false`.
A flag to use fixed positioning of the popover to avoid content being clipped by parental boundaries. Defaults to `true`.

<div style="width: 300px; height: 125px; position: relative; overflow: hidden; z-index: 1; background-color: salmon;">
  <KPop
    title="Look Mah!"
    width="170"
    placement="right"
  >
    <KButton>Click</KButton>
<div class="position-fixed-container">
  <KPop button-text="Fixed positioning" positioning="top">
    <template #content>

>       My parent is too small 😭

Reviewer: 😭

> Boolean to hide close button in popover content.

Reviewer: Is it intended that the content doesn't take the full width when possible? The close button seems to take the full height.

![image](https://github.com/Kong/kongponents/assets/67973710/41790163-cf1b-4124-b9eb-256d0297b22a)

Author: Yes it is intended. We can't make content overflow the close icon in any way so they display side-by-side. This looks better when title is present.

> This is the target element that the popover is appended to. By default its KPop wrapper element.

Reviewer: ```suggestion
This is the target element that the popover is appended to. By default its the KPop wrapper element.
```

> KPop wrapper element. Default value is `div`.

Reviewer: ```suggestion
KPop wrapper element type. Default value is `div`.
```

> List of class names you want to assign to `.k-popover` element.

Reviewer: is it still `k-popover` or did we remove the `k-`?

Author: It's still `k-popover`

> Optional boolean flag to hide popover. Useful for external events. Default value is `false`.

Reviewer: ```suggestion
Optional boolean flag to hide the popover. Useful for external events. Default value is `false`.
```

> Pass a number to use for `z-index` property. Default value is `1000`.

Reviewer: ```suggestion
Pass a number to use for the `z-index` property. Default value is `1000`.
```

  />
  <template #content>
    Must contain at least one special character: *!&#.
  </template>
</KPop>

```html
<KPop
  width="auto"
  hide-close-icon
>
  <KInput
    label="Password"
    type="password"
    placeholder="Enter a strong password"
  />
  <template #content>
    Must contain at least one special character: *!&#.
  </template>
</KPop>
```

### title

Slot for passing custom popover title.

<KPop button-text="Open popover">
  <template #title>
    Popover title
  </template>
  <template #content>
    Popover content.
  </template>
</KPop>

```html
<KPop button-text="Open popover">
  <template #title>
    Popover title
  </template>
  <template #content>
    Popover content.
  </template>
</KPop>
```

### footer

Slot for passing footer content that goes directly underneath main popover content.

<KPop button-text="Open popover">
  <template #content>
    Popover content.
  </template>
  <template #footer>
    <KBadge>Footer</KBadge>
  </template>
</KPop>

```html
<KPop button-text="Open popover">
  <template #content>
    Popover content.
  </template>
  <template #footer>
    <KBadge>Footer</KBadge>
  </template>
</KPop>
```

## Events

### open

> Fires when popover is opened.

Reviewer: ```suggestion
Fires when the popover is opened.
```

### close

> Fires when popover is closed.

Reviewer: ```suggestion
Fires when the popover is closed.
```

### popover-click

> Fires when popover content is clicked.

Reviewer: ```suggestion
Fires when the popover content is clicked.
```

# Popover

> KPop is a popover component that comes in handy when you need to display more content than a tooltip can fit.

Reviewer 2: nit
```suggestion
KPop is a popover component that comes in handy when you need to display more content than can fit in a tooltip.
```

  :data="{ hideCloseIcon: false }"
>
  <div class="vertical-container">
    <KInputSwitch
      v-model="data.hideCloseIcon"
      label="Hide close icon"
    />
    <KPop
      button-text="Open popover"
      :hide-close-icon="data.hideCloseIcon"
    >
      <template #content>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit.
      </template>
    </KPop>
  </div>
</KComponent>

```html
<KPop
  hide-close-icon
  button-text="Open popover"
>
  <template #content>
    Popover content.
  </template>
</KPop>
```

### width

Width of the popover container. Default value is `200px`.

<KPop
  width="500"
  button-text="Open popover"
>
  <template #content>
    Popover content.
  </template>
</KPop>

```html
<KPop
  width="500"
  button-text="Open popover"
>
  <template #content>
    Popover content.
  </template>
</KPop>
```

### maxWidth

Maximum width of the popover container. Default value is `auto`.

<KPop
  max-width="120"
  button-text="Open popover"
>
  <template #content>
    Popover content.
  </template>
</KPop>

```html
<KPop
  max-width="120"
  button-text="Open popover"
>
  <template #content>
    Popover content.
  </template>
</KPop>
```

### target

> This is the target element that the popover is appended to. By default its the KPop wrapper element.

Reviewer 2: ```suggestion
This is the target element selector that the popover is appended to. By default its the KPop wrapper element.

Example: `#my-element-id`
```
